### DISCUSSION TITLE
Labeling "updated" nodes in machine config pool

### DISCUSSION THREAD

**Discussion by SISheogorath** [okd-project/okd#1899] [Ideas]
In order to optimise workloads for OpenShift updates, it would be helpful if one could define affinities to make Pods jump to updated nodes instead of being shuffled multiple times. It would also allow to adjust the scheduler preferences to prefer scheduling workloads on updated nodes, smoothing the update process by reducing potential blocking by PDBs.

Main question would be which controller will do it. The machine-config-operator is of course the prime candidate to attach and remove these labels from node objects along with maintaining the annotations for desired-machineconfig states. But maybe someone has a better idea.

**Comment by vrutkovs** [comment]
We already have `UpdateInProgress: PreferNoSchedule` taint - https://github.com/openshift/machine-config-operator/pull/2686 - to make sure new pods are scheduled on upgraded nodes first

**Reply by SISheogorath** [reply]
I see, I didn't think of taints there. Good point, I guess then it's already a solved problem.

One difference is of course that Pods have less control over their preference e.g. They can't use required affinities for up-to-date nodes and delay their own start (e.g. by waiting for autoscaling or updates to provide the space).

**Reply by vrutkovs** [reply]
Right, perhaps there's an are for improvement here. I don't think labels is a good fit here, but worth a try.

However it first needs to be filed for OCP, I think its best to start with a support request (even if you have a developer account) or filing a bug for https://issues.redhat.com/projects/OKD/issues
